src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -18,58 +18,60 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal enum AnalyzerLoadOption
+    internal sealed partial class AnalyzerAssemblyLoader
     {
-        /// <summary>
-        /// Once the assembly path is chosen, load it directly from disk at that location
-        /// </summary>
-        LoadFromDisk,
+        internal static IAnalyzerAssemblyResolver DiskAnalyzerAssemblyResolver => DiskResolver.Instance;
+        internal static IAnalyzerAssemblyResolver StreamAnalyzerAssemblyResolver => StreamResolver.Instance;
 
-        /// <summary>
-        /// Once the assembly path is chosen, read the contents of disk and load from memory
-        /// </summary>
-        /// <remarks>
-        /// While Windows supports this option it comes with a significant performance penalty due
-        /// to anti virus scans. It can have a load time of 300-500ms while loading from disk 
-        /// is generally 1-2ms. Use this with caution on Windows.
-        /// </remarks>
-        LoadFromStream
-    }
-
-    internal partial class AnalyzerAssemblyLoader
-    {
-        private readonly AssemblyLoadContext _compilerLoadContext;
         private readonly Dictionary<string, DirectoryLoadContext> _loadContextByDirectory = new Dictionary<string, DirectoryLoadContext>(StringComparer.Ordinal);
-        private readonly AnalyzerLoadOption _loadOption;
 
-        internal AssemblyLoadContext CompilerLoadContext => _compilerLoadContext;
-        internal AnalyzerLoadOption AnalyzerLoadOption => _loadOption;
+        public IAnalyzerAssemblyResolver CompilerAnalyzerAssemblyResolver { get; }
+        public AssemblyLoadContext CompilerLoadContext { get; }
+        public ImmutableArray<IAnalyzerAssemblyResolver> AnalyzerAssemblyResolvers { get; }
 
-        internal AnalyzerAssemblyLoader(ImmutableArray<IAnalyzerAssemblyResolver> externalResolvers)
-            : this(null, AnalyzerLoadOption.LoadFromDisk, externalResolvers)
+        internal AnalyzerAssemblyLoader()
+            : this([], [DiskAnalyzerAssemblyResolver], null)
         {
         }
 
-        internal AnalyzerAssemblyLoader(AssemblyLoadContext? compilerLoadContext, AnalyzerLoadOption loadOption, ImmutableArray<IAnalyzerAssemblyResolver> externalResolvers)
+        /// <summary>
+        /// Create a new <see cref="AnalyzerAssemblyLoader"/> with the given resolvers.
+        /// </summary>
+        /// <param name="pathResolvers"></param>
+        /// <param name="assemblyResolvers"></param>
+        /// <param name="compilerLoadContext">This is the <see cref="AssemblyLoadContext"/> where the compiler resides. This parameter
```

Comment:
might want to doc what public property will be set to if this is *not* passed in.